android/Staccato_AN/app/src/main/java/com/on/staccato/presentation/common/location/LocationPermissionManager.kt

```diff
@@ -0,0 +1,86 @@
+package com.on.staccato.presentation.common.location
+
+import android.Manifest
+import android.app.Activity
+import android.content.Context
+import android.content.pm.PackageManager
+import android.view.View
+import androidx.activity.result.contract.ActivityResultContracts
+import androidx.appcompat.app.AppCompatActivity
+import androidx.core.app.ActivityCompat
+import androidx.core.content.ContextCompat
+import androidx.fragment.app.FragmentManager
+import com.on.staccato.R
+import com.on.staccato.presentation.util.showSnackBar
+import dagger.hilt.android.qualifiers.ApplicationContext
+import javax.inject.Inject
+
+class LocationPermissionManager
+    @Inject
+    constructor(
+        @ApplicationContext private val context: Context,
+        private val locationManager: LocationManager,
+    ) {
+        private val locationDialog = LocationDialogFragment()
+
+        fun requestPermissionLauncher(
+            activity: Activity,
```

Comment:
지금은 `Activity`를 인자로 받아 타입 캐스팅을 하면서 `AppCompatActivity`만을 지원하고 있어요.
그런데 `registerForActivityResult()`는 `AppCompatActivity`와 `Fragment` 두 곳에서 호출할 수 있고, `AppCompatActivity`와 `Fragment`는 공통 인터페이스인 [`ActivityResultCaller`](https://developer.android.com/training/basics/intents/result?hl=ko#separate)로 추상화 되어있습니다!
```suggestion
            activityResultCaller: ActivityResultCaller,
```
따라서 파라미터를 위처럼 수정한다면, 추후 `requestPermissionLauncher()`를 `Fragment`에서 호출하는 경우도 대비할 수 있을 것 같아요~

Reply:
```kotlin
    fun requestPermissionLauncher(
        activityResultCaller: ActivityResultCaller,
        activity: Activity,
        view: View,
        actionWhenHavePermission: () -> Unit,
    ) = activityResultCaller.registerForActivityResult(ActivityResultContracts.RequestMultiplePermissions()) { permissions ->
        permissions.forEach { (_, isGranted) ->
            if (isGranted) {
                view.showSnackBar(context.resources.getString(R.string.maps_location_permission_granted_message))
                gpsManager.checkLocationSetting(context, activity, actionWhenHavePermission)
            } else {
                view.showSnackBar(context.resources.getString(R.string.all_location_permission_denial))
            }
        }
    }
```

오!!! 너무 좋은 리뷰네용~ 감사합니다!
`activityResultCaller`를 활용하도록 변경했습니다. 다만 `activity`는 LocationSetting(GPS) 설정 시스템 권한 창을 띄울 때 필요해서 제거하지 않았습니다~

`반영 커밋`: [e0bc98c](https://github.com/woowacourse-teams/2024-staccato/pull/663/commits/e0bc98cbcf13cdd80d7d11ec5c3732bc4179b623)